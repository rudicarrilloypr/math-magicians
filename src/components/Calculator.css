.calculator {
  display: grid;
  grid-template-columns: repeat(4, 50px);
  gap: 1px;
  width: 200px;
  margin: auto;
}

.result {
  grid-column: span 4;
  background: #808080;
  color: #fff;
  padding: 20px;
  font-size: 24px;
  text-align: right;
  height: 20px;
}

button {
  padding: 10px;
  font-size: 24px;
  border: none;
  outline: none;
}

button[data-type="number"] {
  background: #ebe5e0;
  color: #000;
}

button[data-type="operation"] {
  background: #f5923e;
  color: #fff;
}

button[data-type="function"] {
  background: #ebe5e0;
  color: #000;
}

button[data-value="0"] {
  grid-column: span 2;
}
