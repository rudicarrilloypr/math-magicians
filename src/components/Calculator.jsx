import React, { useState } from 'react';
import Button from './Button';
import calculate from '../logic/calculate.js';
import './Calculator.css';

function Calculator() {
  const [total, setTotal] = useState(null);
  const [next, setNext] = useState(null);
  const [operation, setOperation] = useState(null);
  const buttons = ['AC', '+/-', '%', 'รท', 7, 8, 9, 'x', 4, 5, 6, '-', 1, 2, 3, '+', 0, '.', '='];

  const handleClick = (buttonName) => {
    const result = calculate({ total, next, operation }, buttonName);
    setTotal(result.total);
    setNext(result.next);
    setOperation(result.operation);
  };

  const display = (total || '') + (operation || '') + (next || '');

  return (
    <div className="calculator">
      <div className="result">{display || '0'}</div>
      {buttons.map((buttonValue) => (
        <Button key={buttonValue} value={buttonValue} handleClick={handleClick} />
      ))}
    </div>
  );
}

export default Calculator;
