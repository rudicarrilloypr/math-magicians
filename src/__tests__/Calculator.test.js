import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import Calculator from '../components/Calculator';

describe('Calculator', () => {
  it('renders Calculator correctly', () => {
    const { asFragment } = render(<Calculator />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('updates display when a number button is clicked', () => {
    render(<Calculator />);
    fireEvent.click(screen.getByText('1'));
    expect(screen.getByText('1')).toBeInTheDocument();
  });
});
