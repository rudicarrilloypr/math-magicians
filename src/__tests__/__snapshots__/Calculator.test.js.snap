// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator renders Calculator correctly 1`] = `
<DocumentFragment>
  <div
    class="title"
  >
    <h2>
      Let's do some math!
    </h2>
  </div>
  <div
    class="calculator"
  >
    <div
      class="result"
    >
      0
    </div>
    <button
      data-type="function"
      data-value="AC"
      type="button"
    >
      AC
    </button>
    <button
      data-type="function"
      data-value="+/-"
      type="button"
    >
      +/-
    </button>
    <button
      data-type="function"
      data-value="%"
      type="button"
    >
      %
    </button>
    <button
      data-type="operation"
      data-value="รท"
      type="button"
    >
      รท
    </button>
    <button
      data-type="number"
      data-value="7"
      type="button"
    >
      7
    </button>
    <button
      data-type="number"
      data-value="8"
      type="button"
    >
      8
    </button>
    <button
      data-type="number"
      data-value="9"
      type="button"
    >
      9
    </button>
    <button
      data-type="operation"
      data-value="x"
      type="button"
    >
      x
    </button>
    <button
      data-type="number"
      data-value="4"
      type="button"
    >
      4
    </button>
    <button
      data-type="number"
      data-value="5"
      type="button"
    >
      5
    </button>
    <button
      data-type="number"
      data-value="6"
      type="button"
    >
      6
    </button>
    <button
      data-type="operation"
      data-value="-"
      type="button"
    >
      -
    </button>
    <button
      data-type="number"
      data-value="1"
      type="button"
    >
      1
    </button>
    <button
      data-type="number"
      data-value="2"
      type="button"
    >
      2
    </button>
    <button
      data-type="number"
      data-value="3"
      type="button"
    >
      3
    </button>
    <button
      data-type="operation"
      data-value="+"
      type="button"
    >
      +
    </button>
    <button
      data-type="number"
      data-value="0"
      type="button"
    >
      0
    </button>
    <button
      data-type="function"
      data-value="."
      type="button"
    >
      .
    </button>
    <button
      data-type="function"
      data-value="="
      type="button"
    >
      =
    </button>
  </div>
</DocumentFragment>
`;
